### PR TITLE
[GraphBolt] Add `force_reload` to OnDiskDataset.

### PR DESCRIPTION
**Pull request by yxy235**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py [hunk 1]
```diff
@@ -1531,6 +1531,54 @@ def test_OnDiskDataset_preprocess_yaml_content_windows():
             )
 
 
+def test_OnDiskDataset_preprocess_force_reload(capsys):
```
Comment (Rhett-Ying) [inline tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py:1534]: add dedicated testcase for `OnDiskDataset()` which is public API besides `preprocess_ondisk_dataset`(it's internal).

File: python/dgl/graphbolt/impl/ondisk_dataset.py [hunk 2]
```diff
@@ -34,7 +35,9 @@
 
 
 def preprocess_ondisk_dataset(
-    dataset_dir: str, include_original_edge_id: bool = False
+    dataset_dir: str,
+    include_original_edge_id: bool = False,
+    force_reload: bool = False,
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/impl/ondisk_dataset.py:40]:
```suggestion
    force_preprocess: bool = False,
```

File: python/dgl/graphbolt/impl/ondisk_dataset.py [hunk 3]
```diff
@@ -62,13 +67,19 @@ def preprocess_ondisk_dataset(
...
+    processed_dir_prefix = "preprocessed"
+    preprocess_metadata_path = os.path.join(
+        processed_dir_prefix, "metadata.yaml"
+    )
     if os.path.exists(os.path.join(dataset_dir, preprocess_metadata_path)):
-        print("The dataset is already preprocessed.")
-        return os.path.join(dataset_dir, preprocess_metadata_path)
+        if force_reload:
+            shutil.rmtree(os.path.join(dataset_dir, processed_dir_prefix))
+            print("The preprocessed dataset has been removed.")
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/impl/ondisk_dataset.py:77]: the log is a bit confusing. just notice user that we're re-preprocessing, so we removed previously preprocessed one or existing one.

Comment (yxy235) [reply python/dgl/graphbolt/impl/ondisk_dataset.py:77]: Will this log be better?
```
The on-disk dataset is re-preprocessing, so the existing preprocessed dataset has been removed.
```

**Review by Rhett-Ying — APPROVED** [review]

**Pull request merged**